.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,93 @@
+name: Trigger image scanning on pull request and changes to pr
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+    branches: [main]
+
+jobs:
+  deploy:
+    name: Trigger image scanning
+    runs-on: ubuntu-latest
+    env:
+      PEAK_USER_ID: 8877
+      BRANCH: ${{ github.head_ref }}
+      SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+      GH_PAGES_BASE_URL: https://peak-ai.github.io/platform-global-images
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get changed directories
+        id: changed-directories
+        uses: tj-actions/changed-files@v23
+        with:
+          files_ignore: |
+            *.md
+            knowledge-base
+            .github/workflows
+            .gitignore
+            license
+          dir_names: true
+
+      - name: List all changed directories/images
+        run: |
+          for directory in ${{ steps.changed-directories.outputs.all_changed_files }}; do
+            echo "$directory was changed"
+          done
+
+      - name: Run docker build on changed images
+        if: steps.changed-directories.outputs.any_changed == 'true'
+        run: |
+          for directory in ${{ steps.changed-directories.outputs.all_changed_files }}; do
+            tag=$(echo $directory | tr / -)
+            echo "Building $directory"
+            docker build $directory --build-arg PEAK_USER_ID=$PEAK_USER_ID -t $tag
+          done
+
+      - name: Run snyk scan on changed images
+        if: steps.changed-directories.outputs.any_changed == 'true'
```

Comment:
@saurabh0402 should there be a check to ensure that the images (which were changed/edited) were actually build in the previous step ?
Because if they are not successfully build , we might also fail in scanning them.

Reply:
Added.